monitoring/notification/alerts.yaml

```diff
@@ -0,0 +1,105 @@
+x-inputs:
+- name: namespace
+  type: constant
+  value: zenko
+- name: job_notification_producer
+  type: config
+  value: artesca-data-backbeat-notification-producer-headless
+- name: job_notification_processors
+  type: config
+  value: artesca-data-backbeat-notification-processor
+- name: configured_notification_targets
+  type: config
+- name: bucket_notification_enabled
+  type: config
+  value: 0
+- name: notificationFailuresWarningThreshold
+  type: config
+  value: 0.1
+- name: notificationFailuresCriticalThreshold
+  type: config
+  value: 1
+- name: notification_producer_replicas
+  type: config
+  value: 1
+- name: notification_processor_replicas
+  type: config
+  value: 1
+
+groups:
+- name: Bucket Notification
+  rules:
+
+  - alert: NotificationProducerDegraded
+    Expr: |
+      ${bucket_notification_enabled} == bool 1
+        * (sum(up{namespace="${namespace}",job="${job_notification_producer}"})
+        < ${notification_producer_replicas})
+    For: "30s"
+    Labels:
+     severity: warning
+    Annotations:
+      zenko_service: backbeat-notification-producer
+      description: "Less than 100% of notification producers are up and healthy"
+      summary: "Notification producer service is degraded"
+
+  - alert: NotificationProducerCritical
+    Expr: |
+      ${bucket_notification_enabled} == bool 1
+        * (sum(up{namespace="${namespace}",job="${job_notification_producer}"}) * 2
+        < ${notification_producer_replicas})
+    For: "30s"
+    Labels:
+      severity: critical
+    Annotations:
+      zenko_service: backbeat-notification-producer
+      description: "Less than 50% of notification producers are up and healthy"
+      summary: "Notification producer service is in critical state"
+
+  - alert: NotificationProcessorsDegraded
+    Expr: |
+      ${bucket_notification_enabled} == bool 1 
+        * (sum(up{namespace="${namespace}",job=~"${job_notification_processors}.*"}) 
+        < (${configured_notification_targets} * ${notification_processor_replicas}))
```

Comment:
we do not really care about the aggregate, this alert should be generated for each `target` : e.g. use `sum() by(target)` so that there is one alert for each target (which just checks for notification_processor_replicas, and does not use configured_notification_targets)

the target will be a label of expression. and can then be used in the alert summary/name